src/CombFilter/CombFilterIf.cpp

```diff
@@ -0,0 +1,133 @@
+
+// standard headers
+
+// project headers
+#include "MUSI6106Config.h"
+
+#include "ErrorDef.h"
+#include "Util.h"
+
+#include "CombFilterIf.h"
+#include "CombFilter.cpp"
+
+static const char*  kCMyProjectBuildDate = __DATE__;
+
+
+CCombFilterIf::CCombFilterIf ()://(CombFilterType_t eFilterType, float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels) :
+    m_bIsInitialized(false),
+    m_pCCombFilter(0),
+    m_fSampleRate(0)
+{
+    this->reset();
+//    this->init(eFilterType, fMaxDelayLengthInS, fSampleRateInHz, iNumChannels);
+}
+
+
+CCombFilterIf::~CCombFilterIf ()
+{
+    this->reset();
+}
+
+const int  CCombFilterIf::getVersion (const Version_t eVersionIdx)
```

Comment:
Would be nice to have this PR based on the assignment 1 starter code (either by making a separate branch or updating `main` from upstream), so that the parts of the starter code which did not change are omitted from the diff here.